### PR TITLE
[3.4] fix multimap to vpack agency store

### PR DESCRIPTION
**Pull request by kvahed**

### CODE REVIEW

File: arangod/Agency/Store.cpp [hunk 1]
```diff
@@ -771,8 +782,9 @@ void Store::removeTTL(std::string const& uri) {
 
   if (!_timeTable.empty()) {
     for (auto it = _timeTable.cbegin(); it != _timeTable.cend();) {
+      LOG_DEVEL << it->second << " " << uri;
```
Comment (neunhoef) [inline arangod/Agency/Store.cpp]: Please remove.

File: arangod/Agency/Node.h [hunk 2]
```diff
@@ -328,7 +328,7 @@ class Node {
   std::string _nodeName;                ///< @brief my name
   Node* _parent;                        ///< @brief parent
   Store* _store;                        ///< @brief Store
-  Children _children;                   ///< @brief child nodes
+  mutable Children _children;           ///< @brief child nodes
```
Comment (neunhoef) [inline arangod/Agency/Node.h]: Please remove mutable.

File: arangod/Agency/Node.cpp [hunk 3]
```diff
@@ -393,6 +398,7 @@ bool Node::addTimeToLive(long millis) {
 // remove time to live entry for this node
 bool Node::removeTimeToLive() {
   if (_ttl != std::chrono::system_clock::time_point()) {
+    LOG_DEVEL << uri();
```
Comment (neunhoef) [inline arangod/Agency/Node.cpp]: Please remove.

File: arangod/Agency/Node.cpp [hunk 4]
```diff
@@ -320,17 +320,22 @@ Node const& Node::operator()(std::vector<std::string> const& pv) const {
...
+    auto& children = current->_children;
+    auto  child = children.find(key);
 
-    if (child == children.end() ||
-        (child->second->_ttl != std::chrono::system_clock::time_point() &&
-         child->second->_ttl < std::chrono::system_clock::now())) {
+    if (child != children.end() &&
+        child->second->_ttl != std::chrono::system_clock::time_point() &&
+        child->second->_ttl < std::chrono::system_clock::now()) {
+      child->second->removeTimeToLive();
```
Comment (neunhoef) [inline arangod/Agency/Node.cpp]: Lines 329 and 330 must be removed.

Simpler code:

```
if (child == children.end() || ( time conditions )) { throw; }
current = child->second.get();
```

**Comment by kvahed** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/3585/

**Pull request merged**